Shared/SongReader/Song.swift

```diff
@@ -44,48 +37,58 @@ struct Song: Hashable {
     
 }
 
+extension Song: Hashable {
+    static func == (lhs: Song, rhs: Song) -> Bool {
+        guard lhs.title == rhs.title &&
+                lhs.artist == rhs.artist &&
+                lhs.album == rhs.album &&
+                lhs.cover == rhs.cover else { return false }
+        
+        return true
+    }
+}
```

Comment:
A subset of the properties (`lyrics` and `bookmark`) are excluded here but it looks like you're using the synthesised `hash(into:)` which will use all properties (including `lyrics` and `bookmark`). As per the [Hashable docs](https://developer.apple.com/documentation/swift/hashable):

>Hashing a value means feeding its essential components into a hash function, represented by the `Hasher` type. Essential components are those that contribute to the type’s implementation of `Equatable`. Two instances that are equal must feed the same values to `Hasher` in `hash(into:)`, in the same order.

(I made [HashableByKeyPath](https://github.com/JosephDuffy/HashableByKeyPath/) to avoid this exact issue although here I would just recommend adding `hash(into:)`)

Reply:
I'll add that to the todo and merge